cve_bin_tool/extractor.py

```diff
@@ -77,7 +78,28 @@ def can_extract(self, filename):
     async def extract_file_tar(filename, extraction_path):
         """Extract tar files"""
         with ErrorHandler(mode=ErrorMode.Ignore) as e:
-            await aio_unpack_archive(filename, extraction_path)
+            # Python 3.12 has a data filter we can use in extract
+            # tarfile has this available in older versions as well
+            if hasattr(tarfile, "data_filter"):
+                await aio_unpack_archive(filename, extraction_path, filter="data")
```

Comment:
Why don't you just use something like tarfile.extract_all instead of  asynchronously unpacking and then waiting for the operation to finish?

Reply:
Originally this was so multiple files could be unpacked in parallel., which helped to speed up certain types of scan if they're not disk i/o bound.

But now that you mention it, I"m worried that the filters getting passed through unpack_archive might be a problem on older versions of python and tarfile.extract might be better for backwards compatibility.  I couldn't find evidence either way, so I'll probably change this.

Reply:
(I say "probably" because if changing it makes all our tests super slow I might need to rethink.  But I *will* make the change and see what happens.)

Reply:
What has happened so far: bandit apparently doesn't know about filter='data' so we got a security error about unsafe use of tarfile. 😆  (I checked, they have a bug filed about it)